pcsx2/DEV9/ATA/Commands/ATA_SCE.cpp

```diff
@@ -33,23 +54,43 @@ void ATA::HDD_SCE()
 			return;
 	}
 }
-//Has
-//ATA_SCE_IDENTIFY_DRIVE @ 0xEC
-
-//ATA_SCE_SECURITY_ERASE_PREPARE @ 0xF1
-//ATA_SCE_SECURITY_ERASE_UNIT
-//ATA_SCE_SECURITY_FREEZE_LOCK
-//ATA_SCE_SECURITY_SET_PASSWORD
-//ATA_SCE_SECURITY_UNLOCK
-
-//ATA_SCE_SECURITY_WRITE_ID @ 0x20
-//ATA_SCE_SECURITY_READ_ID @ 0x30
-
+// All games that have ability to install data into HDD now will verify HDD as original sce HDD. Resident Evil: Outbreak File #1 for example.
 void ATA::SCE_IDENTIFY_DRIVE()
 {
 	PreCmd();
 
-	//HDD_IdentifyDevice(); //Maybe?
+	// fill sceSec response with default data
+	memcpy(sceSec, "Sony Computer Entertainment Inc.", 32); // Always this magic header
+	memcpy(sceSec + 0x20, "SCPH-20401", 10); // sometimes this matches HDD model, the rest 6 bytes filles with zeroes, or sometimes with spaces
+	memcpy(sceSec + 0x30, "  40", 4); // or " 120" for PSX DESR, reference for ps2 area size
+
+	sceSec[0x40] = 0; // 4-byte HDD internal SCE serial, does not match real HDD serial
+	sceSec[0x41] = 0;
+	sceSec[0x42] = 0;
+	sceSec[0x43] = 0x01;
+
+	// purpose of next 12 bytes is unknown
+	sceSec[0x44] = 0; // always zero
+	sceSec[0x45] = 0; // always zero
+	sceSec[0x46] = 0x1a;
+	sceSec[0x47] = 0x01;
+	sceSec[0x48] = 0x02;
+	sceSec[0x49] = 0x20;
+	sceSec[0x4a] = 0; // always zero
+	sceSec[0x4b] = 0; // always zero
+	// next 4 bytes always this value
+	sceSec[0x4c] = 0x01;
+	sceSec[0x4d] = 0x03;
+	sceSec[0x4e] = 0x11;
+	sceSec[0x4f] = 0x01;
+	// 0x50 - 0x80 is a random unique block of data
+	// 0x80 and up - zero filled
+
+	// if exists .hddid file along with HDD image, use its content for sceSec response
+	std::string hddid_file(Path::ReplaceExtension(EmuConfig.DEV9.HddFile, "hddid"));
+	auto fp = FileSystem::OpenManagedCFile(hddid_file.c_str(), "rb");
+	if !(std::fread(sceSec, 1, 128, fp.get()))
```

Comment:
You read 128 elements of size 1, so it's probably best rewritten as:
```suggestion
	if (std::fread(sceSec, 1, 128, fp.get()) != 128)
```